avm/res/recovery-services/vault/main.bicep

```
@@ -0,0 +1,422 @@
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource rsv 'Microsoft.RecoveryServices/vaults@2023-01-01' = {
+  name: name
+  location: location
+  tags: tags
+  identity: identity
+  sku: {
+    name: 'RS0'
+    tier: 'Standard'
+  }
+  properties: {
+    monitoringSettings: !empty(monitoringSettings) ? monitoringSettings : null
+    securitySettings: !empty(securitySettings) ? securitySettings : null
+    publicNetworkAccess: publicNetworkAccess
+  }
+}
+
+module rsv_replicationFabrics 'replication-fabric/main.bicep' = [for (replicationFabric, index) in replicationFabrics: {
+  name: '${uniqueString(deployment().name, location)}-RSV-Fabric-${index}'
+  params: {
+    recoveryVaultName: rsv.name
+    name: contains(replicationFabric, 'name') ? replicationFabric.name : replicationFabric.location
+    location: replicationFabric.location
+    replicationContainers: contains(replicationFabric, 'replicationContainers') ? replicationFabric.replicationContainers : []
+  }
+  dependsOn: [
+    rsv_replicationPolicies
+  ]
+}]
+
+module rsv_replicationPolicies 'replication-policy/main.bicep' = [for (replicationPolicy, index) in replicationPolicies: {
+  name: '${uniqueString(deployment().name, location)}-RSV-Policy-${index}'
+  params: {
+    name: replicationPolicy.name
+    recoveryVaultName: rsv.name
+    appConsistentFrequencyInMinutes: contains(replicationPolicy, 'appConsistentFrequencyInMinutes') ? replicationPolicy.appConsistentFrequencyInMinutes : 60
+    crashConsistentFrequencyInMinutes: contains(replicationPolicy, 'crashConsistentFrequencyInMinutes') ? replicationPolicy.crashConsistentFrequencyInMinutes : 5
+    multiVmSyncStatus: contains(replicationPolicy, 'multiVmSyncStatus') ? replicationPolicy.multiVmSyncStatus : 'Enable'
+    recoveryPointHistory: contains(replicationPolicy, 'recoveryPointHistory') ? replicationPolicy.recoveryPointHistory : 1440
+  }
+}]
+
+module rsv_backupStorageConfiguration 'backup-storage-config/main.bicep' = if (!empty(backupStorageConfig)) {
+  name: '${uniqueString(deployment().name, location)}-RSV-BackupStorageConfig'
+  params: {
+    recoveryVaultName: rsv.name
+    storageModelType: backupStorageConfig.storageModelType
+    crossRegionRestoreFlag: backupStorageConfig.crossRegionRestoreFlag
+  }
+}
+
+module rsv_backupFabric_protectionContainers 'backup-fabric/protection-container/main.bicep' = [for (protectionContainer, index) in protectionContainers: {
+  name: '${uniqueString(deployment().name, location)}-RSV-ProtectionContainers-${index}'
+  params: {
+    recoveryVaultName: rsv.name
+    name: protectionContainer.name
+    sourceResourceId: protectionContainer.sourceResourceId
+    friendlyName: protectionContainer.friendlyName
+    backupManagementType: protectionContainer.backupManagementType
+    containerType: protectionContainer.containerType
```

```suggestion
    sourceResourceId: protectionContainer.?sourceResourceId
    friendlyName: protectionContainer.?friendlyName
    backupManagementType: protectionContainer.?backupManagementType
    containerType: protectionContainer.?containerType
```

My bad - should be like the above. Otherwise it's not optional :)

---

made the changes.